pkg/driver/azure-disk/azure_disk.go

```diff
@@ -0,0 +1,376 @@
+			Sidecars: []generator.SidecarConfig{
+				commongenerator.DefaultProvisionerWithSnapshots.WithExtraArguments(
+					"--default-fstype=ext4",
+					"--feature-gates=Topology=true",
+					"--extra-create-metadata=true",
+					"--timeout=60s",
+				),
+				commongenerator.DefaultAttacher.WithExtraArguments(
+					"--timeout=60s",
+				),
+				commongenerator.DefaultResizer.WithExtraArguments(
+					"--timeout=300s",
+				),
+				commongenerator.DefaultSnapshotter.WithExtraArguments(
+					"--timeout=300s",
+					"--extra-create-metadata",
+				),
+				commongenerator.DefaultLivenessProbe.WithExtraArguments(
+					"--probe-timeout=3s",
+				),
+			},
+			Assets: commongenerator.DefaultControllerAssets,
+			AssetPatches: commongenerator.DefaultAssetPatches.WithPatches(generator.HyperShiftOnly,
+				"controller.yaml", "overlays/azure-disk/patches/controller_add_hypershift_controller.yaml",
+			),
+		},
+
+		GuestConfig: &generator.GuestConfig{
+			DaemonSetTemplateAssetName: "overlays/azure-disk/patches/node_add_driver.yaml",
+			LivenessProbePort:          10300,
+			Sidecars: []generator.SidecarConfig{
+				commongenerator.DefaultNodeDriverRegistrar,
+				commongenerator.DefaultLivenessProbe.WithExtraArguments(
+					"--probe-timeout=3s",
+				),
+			},
+			Assets: commongenerator.DefaultNodeAssets.WithAssets(generator.AllFlavours,
+				"overlays/azure-disk/base/csidriver.yaml",
+				"overlays/azure-disk/base/storageclass.yaml",
+				"overlays/azure-disk/base/volumesnapshotclass.yaml",
+				"overlays/azure-disk/base/volumesnapshotclass_ash.yaml",
+			),
+		},
+	}
+}
+
+// GetAzureDiskOperatorConfig returns runtime configuration of the CSI driver operator.
+func GetAzureDiskOperatorConfig() *config.OperatorConfig {
+	return &config.OperatorConfig{
+		CSIDriverName:                   opv1.AzureDiskCSIDriver,
+		UserAgent:                       "azure-disk-csi-driver-operator",
+		AssetReader:                     assets.ReadFile,
+		AssetDir:                        generatedAssetBase,
+		OperatorControllerConfigBuilder: GetAzureDiskOperatorControllerConfig,
+	}
+}
+
+// GetAzureDiskOperatorControllerConfig returns second half of runtime configuration of the CSI driver operator,
+// after a client connection + cluster flavour are established.
+func GetAzureDiskOperatorControllerConfig(ctx context.Context, flavour generator.ClusterFlavour, c *clients.Clients) (*config.OperatorControllerConfig, error) {
+	cfg := operator.NewDefaultOperatorControllerConfig(flavour, c, "AzureDisk")
+
+	infra, err := c.ConfigClientSet.ConfigV1().Infrastructures().Get(ctx, infrastructureName, metaV1.GetOptions{})
+	if err != nil {
+		klog.Errorf("unable to fetch infra object from guest cluster with: %v", err)
+		return nil, fmt.Errorf("unable to fetch infra object: %v", err)
+	}
+
+	insideStackHub := runningInAzureStackHub(infra)
+
+	// We need featuregate accessor made available to the operator pods
```

Comment:
I think it would make sense to move the feature gate code to starter.go / a separate file, IMO we can expect that more operators will use it. `client.Clients` can have a field (accessor?) to read the gates.

Reply:
+1 , I can do that.

Reply:
This has not been addressed yet.